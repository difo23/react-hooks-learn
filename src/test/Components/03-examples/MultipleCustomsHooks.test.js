import { shallow } from 'enzyme'
import React from 'react'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch'
jest.mock('../../../hooks/useFetch.js');
jest.mock('../../../hooks/useCounter');

describe('Test in  MultipleCustomsHooks', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => { }
    });

test('should to show success component ', () => {

    useFetch.mockReturnValue({
        data: null,
        loading: true,
        error: null
    })

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();

})

test('should to test useFetch return', () => {

    useFetch.mockReturnValue({
        data: [{ author: 'Juan', quote: 'Corazon de piedras' }],
        loading: false,
        error: null
    })

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find('.blockquote-footer').text().trim()).toBe('Juan');
    expect(wrapper.find('#quote').text().trim()).toBe('Corazon de piedras')
})


})
