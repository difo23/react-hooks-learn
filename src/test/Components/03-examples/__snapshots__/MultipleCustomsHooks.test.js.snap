// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in  MultipleCustomsHooks should to show success component  1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
